src/heartpredict/backend/data.py

```diff
@@ -0,0 +1,88 @@
+from dataclasses import dataclass
+from functools import lru_cache
+from pathlib import Path
+from typing import Self
+
+import joblib
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler
+
+
+@dataclass
+class NumpyMatrix: 
+    x: np.ndarray
+    y: np.ndarray
+
+
+class ProjectData:
+    def __init__(self, csv: Path) -> None:
+        self.project_path = Path(__file__).parent.parent.parent.parent
+        self.df = pd.read_csv(self.project_path.joinpath(csv))
+
+
+    @classmethod    
+    @lru_cache
+    def build(
+        cls, csv: Path = Path("data/heart_failure_clinical_records.csv")
+        ) -> Self:
+        return cls(csv)
+
+
+class MLData:
+    def __init__(
+        self, project_data: ProjectData, test_size: float, random_seed: int
+    ) -> None:
+        self.project_data = project_data
+        self.test_size = test_size
+        self.random_seed = random_seed
+        self.raw = self._get_raw_matrix(project_data.df)
+        self.train, self.test = self._get_prepared_matrices()
```

Comment:
```suggestion
        self.train, self.validation = self._get_prepared_matrices()
```

Here we split into (training/test) and validation set.